src/shogun2-core/src/main/java/de/terrestris/shogun2/model/layer/source/WmtsLayerDataSource.java

```diff
@@ -63,6 +239,14 @@ public boolean equals(Object obj) {
 
         return new EqualsBuilder().
             appendSuper(super.equals(other)).
+            append(getFormat(), other.getFormat()).
+            append(getRequestEncoding(), other.getRequestEncoding()).
+            append(getWmtsLayer(), other.getWmtsLayer()).
+            append(getMatrixSet(), other.getMatrixSet()).
+            append(getProjection(), other.getProjection()).
+            append(getWmtsStyle(), other.getWmtsStyle()).
+            append(getTileGrid(), other.getTileGrid()).
+            append(getUrls(), other.getUrls()).
```

Comment:
Maybe we should not use getTileGrid and getUrls here as this may lead to "expensive" DB queries?

Reply:
As collections are currently included in almost every model of SHOGun2, I've appended `getTileGrid` and `getUrls` to `equals` and `hashCode` method.

However, IMHO,  we could think about removing these in the models since all models inherit from `PersistentObject` which generates a unique id that can be used for equality check, for example.

Reply:
I don't think that this will make a big difference, so feel free to merge ;-)
I don't think that using the id field from `PersistenObject` is a good choice since it is possible that a non-persisted entity is already managed in a set and will be persisted afterwards, which may lead to unexpected problems (at least in theory :wink: )